tests/OpenApiTranslatorTest.php

```diff
@@ -0,0 +1,319 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Tests;
+
+use PHPUnit\Framework\TestCase;
+use ScrumWorks\OpenApiSchema\OpenApiTranslator;
+use ScrumWorks\OpenApiSchema\ValueSchema\ArraySchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\BooleanSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\EnumSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\FloatSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\HashmapSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\IntegerSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\MixedSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\ObjectSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\StringSchema;
+use ScrumWorks\OpenApiSchema\ValueSchema\ValueSchemaInterface;
+
+class OpenApiTranslatorTest extends TestCase
+{
+    private OpenApiTranslator $translator;
+
+    protected function setUp(): void
+    {
+        $this->translator = new OpenApiTranslator();
+    }
+
+    /**
+     * @dataProvider dpMixedArray
+     * @dataProvider dpIntegerArray
+     * @dataProvider dpFloatArray
+     * @dataProvider dpBooleanArray
+     * @dataProvider dpStringArray
+     * @dataProvider dpEnumArray
+     * @dataProvider dpArrayArray
+     * @dataProvider dpHashmapArray
+     * @dataProvider dpObjectArray
+     */
+    public function test(ValueSchemaInterface $schema, array $expected): void
+    {
+        $this->assertEquals($expected, $this->translator->translateValueSchema($schema));
+    }
+
+    public function dpMixedArray(): array
+    {
+        return [
+            'mixed:minimal' => [new MixedSchema(), []],
+            'mixed:full' => [
+                new MixedSchema(true, 'mixed'),
+                [
+                    'nullable' => true,
+                    'description' => 'mixed',
+                ],
+            ],
+        ];
+    }
+
+    public function dpIntegerArray(): array
+    {
+        return [
+            'integer:minimal' => [
+                new IntegerSchema(),
+                [
+                    'type' => 'integer',
+                    'format' => 'int32',
```

Comment:
Je spravne, ze integer generuje int32? Protoze jiste pouzivame 64 bitove int v php

Reply:
jo, ale JSON pouziva 32bit

Reply:
json nema maximalni velikost integeru nijak omezenou. Javascript ma, ale to je jina pohadka